src/persist-client/src/operators/shard_source.rs

```diff
@@ -593,6 +626,99 @@ where
     (descs_stream, activate_on_drop)
 }
 
+pub(crate) fn granular_backpressure<T, G>(
```

Comment:
very exciting that this can be extracted as a separate operator. i wonder if we could do the same for the other backpressure, too! (not for this PR, of course)